### PR TITLE
Fix VBox VM Startup with 3D Acceleration Enabled

### LINKED ISSUE #313372: Virtualbox is unable to do 3D acceleration
### Describe the bug
VMs with 3D acceleration don't start. I see the following in VirtualBox log:
```
00:00:03.223187 rtldrNativeLoad: dlopen('D3DCompiler_47.so', RTLD_NOW | RTLD_LOCAL) failed: D3DCompiler_47.so: cannot open shared object file: No such file or directory
00:00:03.223700 rtldrNativeLoad: dlopen('libvulkan.so.1', RTLD_NOW | RTLD_LOCAL) failed: libvulkan.so.1: cannot open shared object file: No such file or directory
```

### Steps To Reproduce
1. Enable 3D acceleration in VM

### Expected behavior
VM starts

### Additional context
Setting the path to vulkan-loader's lib directory via LD_LIBRARY_PATH has helped to me. Perhaps the dlopen calls should be patched to contain full paths? D3DCompiler is likely needed just as well for Windows guests via DXVK?

Personally I have the problem with virtualboxKvm, not sure about the common virtualbox.

### Notify maintainers

@svanderburg @friedrichaltheide @blitz

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.7.9-lqx1, NixOS, 24.05 (Uakari), 24.05.19700101.dirty`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/etc/static/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by blitz**
## Description of changes

When 3D acceleration is enabled, VirtualBox needs libvulkan.so to start. This PR adds this to the existing wrapper.

In case, we can't wrap the binary this will not work unfortunately.

Fixes #313372.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by blitz** [comment]
This is confirmed to resolve the issue.

**Comment by FriedrichAltheide** [comment]
I will approve the PR once #303790 is merged

**Review by FriedrichAltheide — APPROVED** [review]

**Review by bjornfor — APPROVED** [review]

**Comment by rolfschr** [comment]
I feel like this fix is the same/similar as/to https://github.com/NixOS/nixpkgs/issues/282717.

**Pull request merged**
